app/models/container_group.rb

```diff
@@ -37,6 +38,17 @@ def ready_condition_status
     ready_condition.try(:status) || 'None'
   end
 
+  def container_states_summary
+    summary = Hash.new(0)
+    containers.each { |container| summary[container.state.to_sym] += 1 }
+    summary
```

Comment:
@simon3z @zakiva I think we should use includes(:containers) when loading the ContainerGroup (checked it, it doesn't create an extra query when calling .containers)
https://coderwall.com/p/utqhiq/mind-the-order-of-includes-for-has_many-through, this is a nice to read post about eager loading that gives an example similar to our situation. 
Using group should work without any problems (although using it will create another query even when using eager loading).